docs/adr/0002-docker-images-for-ruby-applications.md

```diff
@@ -0,0 +1,113 @@
+   for multiple GOV.UK apps could be frustratingly slow with the same Gems
+   installed in each different image, it also resulted in high disk usage.
+3. The images were only exercised in publishing-e2e-tests which ran them
+   [in production mode][production], this meant they didn't necessarily have
+   sufficient dependencies for development usage  (e.g. Google Chrome for UI
+   testing).
+4. The images [embedded the applications files][embed-files] - this meant
+   that by default any edits to files would require the image to be rebuilt.
+
+Thus we considered whether it would be appropriate to modify these images to
+solve the problems or to create different images for GOV.UK Docker.
+
+[Docker images]: https://docs.docker.com/get-started/overview/#images
+[publishing-e2e-tests]: https://github.com/alphagov/publishing-e2e-tests
+[example-dockerfile]: https://github.com/alphagov/publishing-api/blob/8b6bc39a3f6b73c2c3e73def6a1c2a6e7687d789/Dockerfile
+[embed-files]: https://github.com/alphagov/specialist-publisher/blob/33a6f3a5d7c8b5ea79d0cd5a8232ed40391db1ef/Dockerfile#L26
+[gem-install]: https://github.com/alphagov/government-frontend/blob/51d86566de33c122aea20f30bcc0fd0d36572752/Dockerfile#L16
+[production]: https://github.com/alphagov/publishing-e2e-tests/blob/3c0e50f66c2c61a7bb98eb07cacd4fcf6233fb51/docker-compose.yml#L14
+
+## Decision
+
+We decided that we should not reuse the existing images and that we would
+resolve the problems with a different image approach for GOV.UK Docker. While
+some of the problems could be resolved we reasoned that the existing images
+represent [standard industry practice][docker-rails] for web application images
+and that this practice was optimised for running a production instance of an
+application. We felt that the needs of production and development were
+sufficiently distinct that it would be simpler to have separate images than try
+to consolidate.
+
+We instead [created a Dockerfile][base-dockerfile] that is stored in
+GOV.UK Docker and this is used by the majority of GOV.UK Docker Ruby projects.
+This base Dockerfile contains a collection of common dependencies used by a
+significant number of projects. Projects that have additional dependencies
+have their [own Dockerfiles][whitehall-dockerfile]. This allowed us to ensure
+that projects had their necessary development dependencies.
+
+We resolved the problems for Gem installations with a [shared Docker
+mount][home-volume]. Using this allowed the installation, and thus updating,
+of Ruby versions and Gems to be done [outside a Docker build
+process][ruby-install], meaning these tasks can be achieved without rebuilding
+the image. This shared mount is used across projects and allows Ruby and Gem
+dependencies to be reused.
+
+We resolved the problem of embedded files with a [mount for the ~/govuk
+directory][govuk-mount] which is expected to project repositories. This mount
```

Comment:
```suggestion
directory][govuk-mount]. This mount
```

I'm not sure what this meant, and what follows makes sense without it.

Reply:
Ah I missed a bit. It was meant to say "which is expected to store projects' repositories" - I've gone with your removed suggestion.